### PR TITLE
Added try-catch strategy in Reporting section

### LINKED ISSUE #3426: Refactor try catch in Management > Reporting
| Parent Issue |
|:---:|
|#3265| 

**Description**
In order to improve better error handling in the front-end, we should implement a better try-catch strategy in the `Management > Reporting` section, services, controller.

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi team,
This PR implement our new error handling strategy on the Ruleset section

Close #3426

### CODE REVIEW

**Review by gabiwassan — CHANGES_REQUESTED** [review]

File: public/controllers/management/components/management/reporting/reporting-table.js [hunk 1]
```diff
@@ -129,39 +159,38 @@ class WzReportingTable extends Component {
...
+    try {
+      const results = items.map(async (item, i) => {
+        await this.reportingHandler.deleteReport(item.name);
+      });
 
-    Promise.all(results).then(completed => {
-      this.props.updateIsProcessing(true);
-      this.showToast('success', 'Success', 'Deleted successfully', 3000);
-    });
+      Promise.all(results).then((completed) => {
```
Comment (Desvelao) [inline public/controllers/management/components/management/reporting/reporting-table.js:172]: **issue**: if there is some error in this `Promise.all`, the `catch` from the `try/catch` block doesn't capture the error.

**suggestion**: We should use `await`
```js
await Promise.all(results)
this.props.updateIsProcessing(true);
this.showToast('success', 'Success', 'Deleted successfully', 3000);
```

File: public/controllers/management/components/management/reporting/reporting-table.js [hunk 2]
```diff
@@ -81,11 +94,31 @@ class WzReportingTable extends Component {
     const columns = this.reportingColumns.columns;
     const message = isLoading ? null : 'No results...';
 
+    const deleteReport = (itemList) => {
+      try {
+        this.deleteReport(itemList);
```
Comment (Desvelao) [inline public/controllers/management/components/management/reporting/reporting-table.js:99]: **issue**: this method is asynchronous, should we wait until the execution is completed?

File: public/controllers/management/components/management/reporting/reporting-table.js [hunk 3]
```diff
@@ -81,11 +94,31 @@ class WzReportingTable extends Component {
...
+        this.props.updateShowModal(false);
+      }catch(error){
+        const options = {
+          context: `${WzReportingTable.name}.deleteReport`,
+          level: UI_LOGGER_LEVELS.ERROR,
+          severity: UI_ERROR_SEVERITIES.BUSINESS,
+          error: {
+            error: error,
+            message: ` Error deleting report: ${error.message || error}`,
```
Comment (Desvelao) [inline public/controllers/management/components/management/reporting/reporting-table.js:108]: **nitpick**: Remove the space at the beginning of the `message` value.

**Review by gabiwassan — APPROVED** [review]
LGTM! 🚀

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark:

**Pull request merged**